DSL/Query/FunctionScoreQuery.php

```diff
@@ -37,14 +40,19 @@ class FunctionScoreQuery implements BuilderInterface
     private $functions;
 
     /**
-     * @param BuilderInterface $filterOrQuery
+     * @param BuilderInterface $query
      * @param array            $functions
      * @param array            $parameters
+     * @param string           $dslType
      */
-    public function __construct(BuilderInterface $filterOrQuery, array $functions, array $parameters = [])
-    {
-        $this->dslType = array_slice(explode('\\', get_class($filterOrQuery)), -2, 1)[0];
-        $this->filterOrQuery = $filterOrQuery;
+    public function __construct(
+        BuilderInterface $query,
+        array $functions,
+        array $parameters = [],
+        $dslType = self::USE_FILTER
```

Comment:
Maybe default value should be ``self::USE_QUERY`` since it is ``FunctionScoreQuery `` query.